### PR TITLE
SITES-2948 Accessibility - Role: The element's role is missing or incorrect.

### PR DESCRIPTION
**Pull request by StoianLucian**
## Description
https://jira.corp.adobe.com/browse/SITES-2948

Added required role from the specified buttons

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: coral-component-buttongroup/src/scripts/ButtonGroup.js [hunk 1]
```diff
@@ -543,7 +543,13 @@ const ButtonGroup = Decorator(class extends BaseFormField(BaseComponent(HTMLElem
         item.setAttribute('role', 'radio');
         item.setAttribute('tabindex', item.hasAttribute('selected') ? 0 : -1);
       } else {
-        item.setAttribute('role', 'checkbox');
+         if((item.getAttribute('data-action') === '#lists')) {
+            item.setAttribute('role', 'Lists');
```
Comment (majornista) [inline coral-component-buttongroup/src/scripts/ButtonGroup.js:547]: "Lists" is not a valid `role` in WAI-ARIA.

File: coral-component-buttongroup/src/scripts/ButtonGroup.js [hunk 2]
```diff
@@ -543,7 +543,13 @@ const ButtonGroup = Decorator(class extends BaseFormField(BaseComponent(HTMLElem
...
+          } else if ((item.getAttribute('data-action') === 'links#modifylink')) {
+            item.setAttribute('role', 'Hyperlink');
```
Comment (majornista) [inline coral-component-buttongroup/src/scripts/ButtonGroup.js:549]: "Hyperlink" is not a valid `role` in WAI-ARIA.

**Review by majornista — CHANGES_REQUESTED** [review]
I made a comment on the issue: https://jira.corp.adobe.com/browse/SITES-2948?focusedCommentId=34514063&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-34514063

In ButtonGroup, the appropriate role for a button within the group is determined by the `selectionMode` attribute on the `<coral-buttongroup>` element. You can see an examples of the three possible selection modes here: https://opensource.adobe.com/coral-spectrum/examples/#buttongroup. The issue SITES-2948 relates to the coral-richtexteditor, which used an earlier CloudUI version of the coral-buttongroup, where one needed to explicitly add the `selectionMode="none"` to the component instance.

**Comment by StoianLucian** [comment]
@majornista i attached the new PRs since i have to make modification in coralui-component-richetexteditor
[master ](https://git.corp.adobe.com/Coral/coralui-component-richtexteditor/pull/271) 
[6.5](https://git.corp.adobe.com/Coral/coralui-component-richtexteditor/pull/270)

**Pull request closed without merging**